### PR TITLE
[WIP][Console] Set `box` style table as default

### PR DESCRIPTION
**Pull request by maidmaid**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/25456#pullrequestreview-103410254
| License       | MIT
| Doc PR        | /

### CODE REVIEW

**Review by nicolas-grekas — DISMISSED** [review]

**Review by ogizanagi — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Can anyone actually try on Windows? Because here, this is UTF-8, but these require CP437, which we do not output AFAIK.

**Comment by ogizanagi** [comment]
I thought it was [already tested on Windows](https://github.com/symfony/symfony/pull/25456#issuecomment-372977133). But indeed, a confirmation would be great.

File: src/Symfony/Component/Console/Helper/Table.php [hunk 1]
```diff
@@ -130,6 +130,10 @@ public static function getStyleDefinition($name)
      */
     public function setStyle($name)
     {
+        if ('symfony-style-guide' === $name) {
+            @trigger_error(sprintf('Style symfony-style-guide is deprecated since Symfony 4.1.', __METHOD__), E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Helper/Table.php]: The "box" style doesn't work on Windows `cmd.com` terminal, so we cannot deprecate `symfony-style-guide`.

**Review by nicolas-grekas — CHANGES_REQUESTED** [review]
I confirm unicode characters are not supported on `cmd.com`. It works with the xterm one. What could work on both Linux and Windows is using `box` when `preg_match('/utf-?8/i', getenv('LANG'))`.

**Comment by nicolas-grekas** [comment]
No need to deprecate IMHO, the symfony-style-guide is still visually nice.

**Comment by fabpot** [comment]
Ah, I didn't realize that the default Symfony theme is not the console default. I would say that we don't need to change anything then. The current default looks great as is and works everywhere. Sorry for asking for a change, I should have checked what we were using first.

**Comment by maidmaid** [comment]
IMO, the `box` style allows the best readability of a table thanks to full lines. It would be nice to use it when possible (as suggested by @nicolas-grekas).

My new suggestion is to use the `box` style when possible, and use `symfony-style-guide` as a workaround when this first one can't work.

**Comment by fabpot** [comment]
I like consistency. So, I'd recommend to close this PR. Sorry for making you work on this one. My mistake.

**Comment by chalasr** [comment]
👍 to close

**Pull request closed without merging**